alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/action/DescriptorActionApi.java

```diff
@@ -132,4 +114,40 @@ public FieldModel saveConfig(final FieldModel fieldModel) {
     public FieldModel deleteConfig(final FieldModel fieldModel) {
         return fieldModel;
     }
+
+    private void checkRelatedFields(final ConfigField field, final Map<String, ConfigField> descriptorFields, final FieldModel fieldModel, final Map<String, String> fieldErrors) {
+        final Set<String> requiredRelatedFields = field.getRequiredRelatedFields();
+        for (final String relatedFieldKey : requiredRelatedFields) {
+            final ConfigField relatedField = descriptorFields.get(relatedFieldKey);
+            validateAnyRelatedFieldsMissing(relatedField, fieldModel, fieldErrors);
+        }
+        final Set<String> disallowedRelatedFields = field.getDisallowedRelatedFields();
+        for (final String disallowedRelatedFieldKey : disallowedRelatedFields) {
+            final ConfigField relatedField = descriptorFields.get(disallowedRelatedFieldKey);
+            validateAnyDisallowedFieldsSet(relatedField, fieldModel, fieldErrors, field.getLabel());
+        }
+    }
+
+    private Boolean hasValueOrChecked(final FieldValueModel fieldValueModel, final String type) {
+        final Boolean isValueTrue = fieldValueModel.getValue().map(Boolean::parseBoolean).orElse(false);
+        final Boolean isCheckbox = FieldType.CHECKBOX_INPUT.getFieldTypeName().equals(type);
+        return (isValueTrue && isCheckbox) || (!fieldValueModel.containsNoData() && !isCheckbox);
+    }
+
+    private void validateAnyRelatedFieldsMissing(final ConfigField field, final FieldModel fieldModel, final Map<String, String> fieldErrors) {
+        final String key = field.getKey();
+        final Optional<FieldValueModel> optionalFieldValue = fieldModel.getField(key);
+        if (optionalFieldValue.isEmpty() || (optionalFieldValue.isPresent() && optionalFieldValue.get().containsNoData())) {
```

Comment:
You might be able to map the Optional to `containsNoData`. Then you don't need the is present check separate.